### PR TITLE
Only run ARA Analysis if within ARA Perimeter

### PR DESCRIPTION
**Pull request by rajadain**
## Overview

Adds a hand-drawn perimeter of the ARA raster to check the area of interest against when deciding if it should be queried or not. If it is not within the perimeter, `null`s are returned. The shape was drawn by hand as it was the most expedient way to do it:

image

The perimeter is conservative, which is to say that it is smaller than the actual boundary. This is done in the hopes of preempting any edge cases that may query non-existing tiles:

image

This is done without precision. If there is a better way of generating this perimeter from a raster, I'd be up for trying that.

Connects #3209 

### Demo

image

## Testing Instructions

* Check out this branch
* Debug Celery `$ ./scripts/debugcelery.sh`
* Go to a place in the map other than the Mid Atlantic or North East, select an area of interest
    - [x] Ensure that all analyses complete successfully, including Land
* Select an area of interest within the Mid Atlantic or North East region
    - [x] Ensure that all analyses complete successfully, including Land

### CODE REVIEW

**Comment by rajadain** [comment]
To test the perimeter on https://geojson.io/, use [this shape](https://github.com/WikiWatershed/model-my-watershed/blob/tt/gatekeep-ara-raster/src/mmw/mmw/settings/data/ara_perimeter.json) and add this custom map layer:

```
https://{s}.tiles.azavea.com/ara-30m/{z}/{x}/{y}.png
```

File: src/mmw/apps/geoprocessing_api/tasks.py [hunk 1]
```diff
@@ -131,16 +131,20 @@ def analyze_nlcd(result, area_of_interest=None):
...
         total_count += count
         histogram[nlcd] = count + histogram.get(nlcd, 0)
 
     for nlcd, (code, name) in settings.NLCD_MAPPING.iteritems():
         categories.append({
             'area': histogram.get(nlcd, 0) * pixel_width * pixel_width,
             'active_river_area':
-                result.get((nlcd, 1), 0) * pixel_width * pixel_width,
+                result.get((nlcd, 1), 0) * pixel_width * pixel_width
+                if has_ara_results else None,
```
[the remark this answers is not in the document]
Comment (rajadain) [reply src/mmw/apps/geoprocessing_api/tasks.py]: The indentation doesn't pass `flake8`:

```
/opt/app/apps/geoprocessing_api/tasks.py:147:21: E131 continuation line unaligned for hanging indent
```

How about something like this:

```diff
diff --git a/src/mmw/apps/geoprocessing_api/tasks.py b/src/mmw/apps/geoprocessing_api/tasks.py
index b9c31aa7..399ed374 100644
--- a/src/mmw/apps/geoprocessing_api/tasks.py
+++ b/src/mmw/apps/geoprocessing_api/tasks.py
@@ -131,20 +131,21 @@ def analyze_nlcd(result, area_of_interest=None):
     total_count = 0
     categories = []
 
-    has_ara_results = type(result.keys()[0]) == tuple
+    has_ara = type(result.keys()[0]) == tuple
+
+    def area(dictionary, key, default=0):
+        return dictionary.get(key, default) * pixel_width * pixel_width
 
     # Convert results to histogram, calculate total
     for key, count in result.iteritems():
-        nlcd = key[0] if has_ara_results else key
+        nlcd = key[0] if has_ara else key
         total_count += count
         histogram[nlcd] = count + histogram.get(nlcd, 0)
 
     for nlcd, (code, name) in settings.NLCD_MAPPING.iteritems():
         categories.append({
-            'area': histogram.get(nlcd, 0) * pixel_width * pixel_width,
-            'active_river_area':
-                result.get((nlcd, 1), 0) * pixel_width * pixel_width
-                if has_ara_results else None,
+            'area': area(histogram, nlcd),
+            'active_river_area': area(result, (nlcd, 1)) if has_ara else None,
             'code': code,
             'coverage': float(histogram.get(nlcd, 0)) / total_count,
             'nlcd': nlcd,

```

[the remark this answers is not in the document]
Comment (mmcfarland) [reply src/mmw/apps/geoprocessing_api/tasks.py]: Ah, I'll have to defer to the benevolent dictator on this one I guess. I like your proposed refactor, a little DRYer and clearer :+1:

**Comment by rajadain** [comment]
Thanks for reviewing!

**Pull request merged**